### PR TITLE
dev-db/mongodb: add PATH to Environmnet, otherwise build failed w/ clang

### PR DESCRIPTION
**Pull request by clan**
patch had been submitted to upstream, see https://github.com/mongodb/mongo/pull/1600

Closes: https://bugs.gentoo.org/829340

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by clan** [comment]
> Follow up here (regarding your PR); we intentionally do not ingest the environment PATH as this makes build reproducibility and hermiticity much harder.
> Instead, you should set the absolute paths to the tools you want to use on the command line, instead of relying on the PATH.

upstream replied at https://jira.mongodb.org/browse/SERVER-94430 , they won't merge this patch, which might be a Gentoo specificed problem. Is there other elegant way to fix this problem? It seems clang's path not in PATH duing ebuild compiling?

**Comment by Dessa** [comment]
i have something, but im not proud of it, will not apply cleanly since i have local changes but should give you an idea; running test builds at the moment so i can finally get a pr ready for at least a 5.0 bump (i have 6.0 but i don't really want that since 6.0.18 will probably be released next week if i had to guess.)

```
diff --git a/dev-db/mongodb/mongodb-5.0.29.ebuild b/dev-db/mongodb/mongodb-5.0.29.ebuild
index 297b5ba..883e5f6 100644
--- a/dev-db/mongodb/mongodb-5.0.29.ebuild
+++ b/dev-db/mongodb/mongodb-5.0.29.ebuild
@@ -3,6 +3,9 @@

 EAPI=8

+LLVM_COMPAT=( 18 )
+LLVM_OPTIONAL=1
+
 PYTHON_COMPAT=( python3_{10..12} )

 SCONS_MIN_VERSION="3.3.1"
@@ -10,7 +13,7 @@ CHECKREQS_DISK_BUILD="2400M"
 CHECKREQS_DISK_USR="512M"
 CHECKREQS_MEMORY="1024M"

-inherit check-reqs flag-o-matic multiprocessing pax-utils python-any-r1 scons-utils systemd toolchain-funcs
+inherit check-reqs flag-o-matic llvm-r1 multiprocessing pax-utils python-any-r1 scons-utils systemd toolchain-funcs

 MY_PV=r${PV/_rc/-rc}
 MY_P=mongo-${MY_PV}
@@ -77,7 +80,6 @@ PATCHES=(
        "${FILESDIR}/${PN}-4.4.29-no-enterprise.patch"
        "${FILESDIR}/${PN}-5.0.26-boost-1.85.patch"
        "${FILESDIR}/${PN}-5.0.26-boost-1.85-extra.patch"
-       "${FILESDIR}/${PN}-5.0.29-clang.patch"
 )

 python_check_deps() {
@@ -107,6 +109,15 @@ pkg_pretend() {
        fi
 }

+pkg_setup() {
+       if tc-is-clang; then
+               llvm-r1_pkg_setup
+               llvm_fix_tool_path CC CXX
+       else
+               return
+       fi
+}
+
 src_prepare() {
        default

```

**Comment by thesamesam** [comment]
I merged https://github.com/gentoo/gentoo/pull/38551, so this would need a rebase, but I'm confused about the bug here.

I only skimmed https://jira.mongodb.org/browse/SERVER-94430 so far but it seems very odd. You would think that a PATH issue just means it calls itself infinitely, or can't find itself, not being stuck in some random LLVM pass.

What am I missing here? Or were there really several issues going on?

**Comment by thesamesam** [comment]
Ah, wait. OK, this is just for https://bugs.gentoo.org/829340. Now it all makes sense: in Gentoo, we have LLVM slotted and we put `clang` in `/usr/lib/llvm/18/bin` (or whatever), not in `/usr/bin`, and if upstream strip PATH and construct it themselves, they surely won't contain this location.

-> Please rebase then it looks OK to merge.

File: dev-db/mongodb/files/mongodb-5.0.26-scons.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/SConstruct b/SConstruct
```
Comment (thesamesam) [inline dev-db/mongodb/files/mongodb-5.0.26-scons.patch:1]: Please add a link to the upstream bug and also the Gentoo bug at the top of the patch. You may want to include my explanation from https://github.com/gentoo/gentoo/pull/38444#issuecomment-2533699514 too.

See also https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html#patch-descriptions.

**Pull request closed without merging**